cupy/cuda/memory.pyx

```diff
@@ -42,6 +46,55 @@ cdef class Memory:
         return self.ptr
 
 
+cdef class ManagedMemory(Memory):
+
+    """Memory allocation on a CUDA device.
+
+    This class provides an RAII interface of the CUDA memory allocation.
+
+    Args:
+        device (cupy.cuda.Device): Device whose memory the pointer refers to.
+        size (int): Size of the memory allocation in bytes.
+
+    """
+
+    def __init__(self, Py_ssize_t size):
+        self.size = size
+        self.device = None
+        self.ptr = 0
+        if size > 0:
+            self.device = _device.Device()
+            self.ptr = runtime.mallocManaged(size)
+
+    cpdef prefetch(self, stream):
+        """ Prefetch.
+
+        Args:
+            stream (cupy.cuda.Stream): Stream
+
+        """
+        global _cuda_version
+        if _cuda_version is None:
+            _cuda_version = runtime.runtimeGetVersion()
+        if _cuda_version >= 8000 and int(self.device.compute_capability) >= 60:
```

Comment:
This method is using for only performance up.
It is better to do nothing on old CUDA environment.